bsip-00XX Dynamic market fees.md

```diff
@@ -0,0 +1,183 @@
+2. create a new type trade_statistics_object and a new index
+
+
+## Statistics Update
+Statistics updates during fill_order. 
+We need to update statistics for the user and for the specific asset at the same time.
+
+ 
+
+## Calculations
+
+### Average since Day 1
+
+**Trader Activity** is estimated as average trading volume of a user for a given asset for 1 day.
+
+`Trader_activity = total_volume / (Now - first_start_date) //(in days).`
+
+This way we accumulate total bought volume for each user for each asset and average out daily volumes for his entire term of trading of this asset (per day).
+
+Volume discounts are estimated based on the average daily `Trader_Activity  * 30 days.`
+
+_Trade statistics object_
+
+
+```
+class trade_statistics_object
+{
+
+account_id_type owner;
+asset_id_type asset;
+timestamp first_trade_date; //first trade date for this asset
+share_type total_volume; //total bought asset volume
+};
+```
+
+### 30-day sliding window average 
+
+In **trade_statistics_object** we need to store the queue of volumes aggregated by 1 day. In other words, for each asset a user should have a queue of 30  daily volumes.
+
+During the first maintenance interval of the day, yesterday's volume will be added to the queue and the  oldest volume will be deleted (for each asset even for assets where there was no trade done).
+
+This way it will cost significant memory usage!!
+
+Technically we can store only the aggregated sum for 30 days and take the first day volumes from the blockchain. This will take less memory but will take more CPU time.
+
+
+### 30-day volume method
+
+Alternative way would be to store one 30-day volume for each asset for each user.  This way we would only need one 30-day volume  and one aggregated daily volume for today. i.e. only 2 values for each asset.
+
+So the memory requirements will be at least 30 times less than in a previous method.
+
+The algorithm is like this:
+
+```
+For example, the 30-day volume is 90. Hence daily average is 3. 
+Let's say today's volume is 13. 
+First of all, we subtract average daily volume from the 30-day volume.  90-3 =87.
+Then we add today's volume.  So the new 30-day volume would be 87+13 = 100.  
+A new daily average would be 100/30 = 3.33333
+``` 
+
+### 365-day sliding window average 
+
+This is similar to the 30-day sliding window average except the memory requirements for the stored values are SIGNIFICANTLY larger.
+
+## Potential issues
+We need to make sure that the suggested calculations will not lead to the significantly wrong results or the blockchain abuse. 
+We need to consider border conditions: For example, when there is no statistics at all, unusual statistics of many very small orders, very large order at the beginning of the average window and then no activity registered, etc. 
+
+ 
+## Technical challenges
+Some problems include performance as an additional search and update new index are required.
+
```

Comment:
Please add a easy-to-understand Summary for the Shareholders

What are the additional leverages created for BTS holders, what are the benefits? Also, where are the risks?

Reply:
Dynamic market fee is an additional marketing tool to encourage traders. So, this BSIP helps to boost trading volume on BitShares and attract new traders to the ecosystem.